### ISSUE TITLE
Use custom domain for Social

### ISSUE THREAD

**Issue by Tachi107** [nextcloud/social#1049]
**Feature request**
It would be nice if we could use a custom domain instead of the one used by Nextcloud. When people looks at my account from a Mastodon instance, for example, a username like Tachi@cobi.it would look way better than Tachi@nextcloud.cobi.it. 

**Additional context**
Nice:
![tachi nextcloud paoloap ml](https://user-images.githubusercontent.com/34214253/95011853-0bbe0280-0634-11eb-8191-46d61162b5f0.png)

Much better:
![tachi paoloap ml](https://user-images.githubusercontent.com/34214253/95011844-ffd24080-0633-11eb-8c9c-ea9fde33773b.png)

**Comment by JOduMonT** [comment]
Yes this option would be awesome, personally I see 2 cases scenarios

1. more commons; if is we use multiple domain on the same nextcloud instance
2. is the first domain of my nextcloud instance is internal, than I would love to being able to change this for my second domain.

**Comment by spaetz** [comment]
I think we are half-way there, already. The webfinger service responds to all domain requests, so a webfinger against @user@maindomain will point to the instance even when it is installed on a nextcloud.maindomain server. Mastodon has decided not to use DNS SRV entries (https://github.com/tootsuite/mastodon/issues/1931) so you would still need a webfinger support on your maindomain. So ,the one remaining issue is the default display Name (the domain part) on the pages, which would need to be configurable?

**Comment by HammyHavoc** [comment]
This would be sweet.

**Comment by karlitschek** [comment]
Supported — closing.

`social_address` is the hostname accounts are federated under when it is not the host of the Nextcloud installation, which is exactly the `Tachi@cobi.it` on `nextcloud.cobi.it` case here (`lib/Service/ConfigService.php`, documented in `docs/Admin.md` under Configuration):

```
occ config:app:set social social_address --value="example.com"
```

One condition, and the docs say it too: set it **before the first account exists**. An ActivityPub id is what every other server has stored, and changing the address afterwards does not rewrite the ids already issued. The short domain also has to answer `/.well-known/webfinger` for the Nextcloud host, since that is where other servers look; `occ social:check:install` reports whether the addresses line up.
